src/PostgREST/Plan.hs

```diff
@@ -1028,6 +1029,22 @@ callPlan proc ApiRequest{} paramKeys args readReq = FunctionCall {
             | otherwise            -> KeyParams $ specifiedParams [prm]
       prms  -> KeyParams $ specifiedParams prms
 
+-- | Get filter fields/column names from read plan
+getFilterFieldNames :: ReadPlanTree -> [FieldName]
+getFilterFieldNames = foldr (\rp names -> names ++ rpToFieldNames rp) []
+  where
+    rpToFieldNames :: ReadPlan -> [FieldName]
+    rpToFieldNames = logicTreesToFieldName . ReadPlan.where_
+
+    logicTreesToFieldName :: [CoercibleLogicTree] -> [FieldName]
+    logicTreesToFieldName = concatMap coLogicTreeToFieldNames
+
+    coLogicTreeToFieldNames :: CoercibleLogicTree -> [FieldName]
+    coLogicTreeToFieldNames = \case
+      CoercibleStmnt (CoercibleFilter{field=CoercibleField{cfName}}) -> [cfName]
+      CoercibleStmnt (CoercibleFilterNullEmbed _ cfName) -> [cfName]
```

Comment:
I tried `/rpc/getallprojects?select=id,clients(id)&clients.name=not.is.null` to cover this line but it didn't work. Hmm :thinking:.

Reply:
Ah, the filter should be done to the embedding itself, not to a column of the embedding, i.e. replace `clients.name=not.is.null` with `clients=not.is.null`.

Reply:
You may need to adapt it to your PR needs, so instead of replacing it, you could add another one (the current one seems relevant to your changes).